sdk/textanalytics/Azure.AI.TextAnalytics/tests/RecognizeCustomEntitiesTests.cs

```diff
@@ -359,6 +359,8 @@ private void ValidateInDocumentResult(CategorizedEntityCollection entities, List
                 {
                     Assert.IsNotEmpty(entity.SubCategory);
                 }
+
+                Assert.IsNotNull(entity.Resolutions);
```

Comment:
are resolutions part of the custom entity recognition feature too?

Reply:
According to the swagger spec, NER and custom NER share the same type "EntityWithResolution", which is called `CategorizedEntity` in the .NET library. However, custom NER never actually returns any resolutions, so in the case the property is always an empty collection (but not null, which is what I'm checking here).

Reply:
should it be null though?

Reply:
As per the .NET framework design guidelines, we shouldn't return null values for collection properties, and to return an empty collection instead.

https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/guidelines-for-collections

Reply:
I agree with the rule in general, but this seems like a special case. Maybe an empty collection signals to the customer it could also be non-empty? Would returning null here means do not even expect to have a collection empty or not?

Reply:
My feeling is that the fact that the property exists at all signals to the customer that it serves some purpose, which in the case of custom NER, it unfortunately doesn't. I think that returning null wouldn't be enough to decisively override this assumption and worry that instead it could add to the confusion because users can tell it's different but not why ("is this special in some way or is this just a bug?"). Ideally, NER and custom NER would return different types, but this is not possible without a breaking change. For the record, I also have this question in my notes to discuss during the arch review.

Reply:
On another thought, could the entity class stay the same and add one that inherits from it to be used with the prebuilt entity recognition result only?